src/lib/stateBasedTestHelper.ts

```diff
@@ -19,7 +19,7 @@ export function calls<P extends any[], T>(
 }
 
 export function spawns<State, P extends any[], T>(
-  effectOrEffectCreator: BoundEffect<SagaEnvironment<State>, P, T> | FuncWithEnv<State, P, T>,
+  effectOrEffectCreator: BoundFunc<State, P, T>,
```

Comment:
I think `FuncWithEnv` was not ideal, but `BoundFunc ` even less so. It was more `FuncRequiringEnv`, as it isn't bound to any env yet.

Also can you please change the argument name please?

Reply:
What do you think we should call the type?